internal/install/recipe_installer_guided.go

```diff
@@ -0,0 +1,299 @@
+	if err != nil {
+		return err
+	}
+
+	log.WithFields(log.Fields{
+		"possible_matches": len(logMatches),
+	}).Debug("filtered log matches")
+
+	var acceptedLogMatches []types.LogMatch
+	var ok bool
+	for _, match := range logMatches {
+		ok, err = i.userAcceptsLogFile(match)
+		if err != nil {
+			return err
+		}
+
+		if ok {
+			acceptedLogMatches = append(acceptedLogMatches, match)
+		}
+	}
+
+	log.WithFields(log.Fields{
+		"matches": acceptedLogMatches,
+	}).Debug("matches accepted")
+
+	// The struct to approximate the logging configuration file of the Infra Agent.
+	type loggingConfig struct {
+		Logs []types.LogMatch `yaml:"logs"`
+	}
+
+	r.AddVar("DISCOVERED_LOG_FILES", loggingConfig{Logs: acceptedLogMatches})
+
+	_, err = i.executeAndValidateWithProgress(m, r)
+	return err
+}
+
+func (i *RecipeInstaller) fetchRecommendations(m *types.DiscoveryManifest) ([]types.Recipe, error) {
+	log.Debug("fetching recommended recipes")
+
+	recommendations, err := i.recipeFetcher.FetchRecommendations(utils.SignalCtx, m)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving recipe recommendations: %s", err)
+	}
+
+	recommendations = i.filterRecommendations(recommendations)
+
+	if log.IsLevelEnabled(log.DebugLevel) {
+		names := []string{}
+		for _, r := range recommendations {
+			names = append(names, r.Name)
+		}
+
+		log.WithFields(log.Fields{
+			"names":        names,
+			"recipe_count": len(recommendations),
+		}).Debug("recommended integrations")
+	}
+
+	return recommendations, nil
+}
+
+// Filter out infra and logging recipes from recommendations, since they are
+// handled explicitly elsewhere.  This avoids duplicate installation.
+func (i *RecipeInstaller) filterRecommendations(recipes []types.Recipe) []types.Recipe {
+	filteredRecommendations := []types.Recipe{}
+	for _, r := range recipes {
+		if r.Name == infraAgentRecipeName || r.Name == loggingRecipeName {
+			log.WithFields(log.Fields{
+				"name": r.Name,
+			}).Debug("skipping redundant recipe")
+
+			continue
+		}
+
+		filteredRecommendations = append(filteredRecommendations, r)
+	}
+
+	return filteredRecommendations
+}
+
+func (i *RecipeInstaller) userAcceptsLogFile(match types.LogMatch) (bool, error) {
+	msg := fmt.Sprintf("Files have been found at the following pattern: %s Do you want to watch them?", match.File)
+	return i.userAccepts(msg)
+}
+
+func (i *RecipeInstaller) recipeInRecipes(recipe types.Recipe, recipes []types.Recipe) bool {
+	for _, r := range recipes {
+		if recipe.Name == r.Name {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (i *RecipeInstaller) removeRecipes(recipes []types.Recipe, remove ...types.Recipe) []types.Recipe {
+	filtered := []types.Recipe{}
+	for _, recipe := range recipes {
+		for _, r := range remove {
+			if recipe.Name != r.Name {
+				filtered = append(filtered, recipe)
+			}
+		}
+	}
+
+	return filtered
+}
+
+func (i *RecipeInstaller) filterIntegrations(recommendedIntegrations []types.Recipe) ([]types.Recipe, error) {
```

Comment:
Even though this method is private, how do you feel about a doc string?

Reply:
👍